tests/ui/wildcard_imports.rs

```diff
@@ -68,6 +68,31 @@ mod struct_mod {
     }
 }
 
+// issue 9942
+mod underscore_mod {
+    mod exports_underscore {
+        pub use std::ops::Deref as _;
+        pub fn dummy() {}
+    }
+
+    mod exports_underscore_ish {
+        pub use std::ops::Deref as _Deref;
+        pub fn dummy() {}
+    }
+
+    fn does_not_lint() {
+        use self::exports_underscore::*;
```

Comment:
I want to make sure, that I understand this change correctly. This case is allowed, since the module contains an underscore import, which can't be imported apart from a glob import. Is this understanding correct?

Reply:
To my understanding, `use self::exports_underscode::*` is interpreted as follows:

```rust
use self::exports_underscore::{_, dummy};
```

The code imports Deref trait with the name `_`, which is reserved identifier, so clippy fix will emit invalid code.

---

<small>
I'm sorry for not being able to respond for a while due to health issues. Now that I have recovered, I hope to be active again.
</small>

Reply:
Hey @koka831 nice to have you back! There is no need to apologize, health and real life should always come first! It's good to hear that you're doing better again! :D

This makes sense, in that case we can merge this version. Let's see if the branch is still close enough to master, to make bors happy :) (I love this way of testing)